python/lib/modeldata/dmod/modeldata/data/dataset.py

```diff
@@ -577,6 +593,10 @@ def add_data(self, dataset_name: str, dest: str, data: Optional[bytes] = None, s
         """
         pass
 
+    @abstractmethod
+    def combine_partials_into_composite(self, dataset_name: str, item_name: str, combined_list: List[str]) -> bool:
```

Comment:
Not a big deal, but I first thought this had to do with partial python functions, not partial datasets. I wonder if I'm the only one who'd be tripped up or if it might benefit the codebase to come up with a different name like 'combine_partial_datasets' or something. Or I could be even more wrong and it might just need a docstring.

Again, not a showstopper, just something that stuck out to me.